spec/helpers/rpc_mock/server.lua

```diff
@@ -0,0 +1,187 @@
+local helpers = require("spec.helpers")
+local client = require("spec.helpers.rpc_mock.client")
+local default_cert = require("spec.helpers.rpc_mock.default").default_cert
+
+local _M = {}
+local _MT = { __index = _M, }
+
+function _M.new(opts)
+  opts = opts or {}
+  opts.prefix = opts.prefix or "servroot_rpc_tap"
+  opts.role = "control_plane"
+  opts.plugins = "bundled,rpc-proxy"
+  opts.listen = opts.listen or 8005 -- default clustering port
+  opts.cluster_listen = opts.cluster_listen or ("0.0.0.0:" .. opts.listen)
+  opts.mocks = opts.mocks or {}
+  opts.prehooks = opts.prehooks or {}
+  opts.posthooks = opts.posthooks or {}
+  opts.cluster_rpc = "on"
+  opts.cluster_rpc_sync = opts.cluster_rpc_sync or "on"
+  if opts.interception == nil then
+    opts.interception = true
+  end
+
+  for k, v in pairs(default_cert) do
+    if opts[k] == nil then
+      opts[k] = v
+    end
+  end
+
+  return setmetatable(opts, _MT)
+end
+
+function _M.start(self)
+  local bp, db = helpers.get_db_utils(strategy, nil, { "rpc-proxy" })
+
+  local plugin = db.plugins:insert({
+    name = "rpc-proxy",
+    config = {},
+  })
+
+  assert(helpers.start_kong(self))
+
+  self.proxy_client = client.new({
+    cluster_control_plane = self.cluster_listen,
+  })
+
+  if self.interception then
+    self:enable_inception()
+  end
+
+  self.proxy_client.callbacks:register("kong.rpc.proxy.mock", function(proxy_id, proxy_payload)
```

Comment:
mock cp will rpc call this callback to forward `[the RPC call request from tested DP]` to `[client]`